### PR TITLE
Fixes for upgrade tests 2.0.4 -> 2.1

### PR DESCRIPTION
**Pull request by lindnerby**
**Description**

Changes proposed in this pull request:

- Install Kyma CLI in upgrade scenarios with corresponding CLI version, provide kyma install without installed check
- Run migration script when upgrading from 2.0.4 to main
- Introduce env var for fast-integration-tests to skip certain tests for major upgrade 1 to 2 scenario

**Related issue(s)**
* https://github.com/kyma-project/kyma/issues/12692

### CODE REVIEW

File: prow/scripts/lib/kyma.sh [hunk 1]
```diff
@@ -136,6 +136,33 @@ kyma::install_cli() {
         log::info "Kyma CLI is already installed: $(kyma version -c)"
     fi
 }
+
+kyma::force_install_cli() {
```
[the remark this answers is not in the document]
Comment (lindnerby) [reply prow/scripts/lib/kyma.sh]: Yes, I was also preferring that. Changed.

**Review by chrkl — APPROVED** [review]

**Pull request merged**